rfcs/framework-0072-pluggable-transfers.md

```diff
@@ -0,0 +1,507 @@
+
+#### Usage
+
+Uppy integration could be enabled by registering it from a [Webpack Files UI entrypoint](#webpack-files-ui-entrypoint) and then setting it to be used from the site config:
+```python
+FILES_UI_ENTRYPOINT = "invenio-app-rdm-files-uppy"
+```
+
+#### UI workflow
+
+Default dashboard state:
+![Screenshot 2025-02-13 at 8 57 20](https://github.com/user-attachments/assets/85319625-7510-4482-b314-64eac835fc73)
+
+Files chosen for upload:
+![Screenshot 2025-02-13 at 9 07 20](https://github.com/user-attachments/assets/eb1810d5-fd27-4612-9de7-ad028c78c1ca)
+
+Multipart upload in progress:
+![Screenshot 2025-02-13 at 9 07 38](https://github.com/user-attachments/assets/c7be5987-7848-4322-9baf-af8cd9a28e9b)
+
+Upload complete:
+![Screenshot 2025-02-13 at 9 07 55](https://github.com/user-attachments/assets/a5820012-a16a-4270-97ee-82d1392ad948)
+
+Add one more file that fails to upload:
+![Screenshot 2025-02-13 at 9 08 58](https://github.com/user-attachments/assets/252826f3-fb66-44b2-a211-4a72bb0d2564)
+
+## Other Examples
+
+Another potential use case is a specialized transport for handling
+sensitive data in the Crypt4GH file format before storing it in the
+repository (managing encryption keys, etc.).
+
+## How we teach this
+
+A [file storage documentation](https://github.com/inveniosoftware/docs-invenio-rdm/blob/master/docs/reference/file_storage.md)
+page will be updated to reflect these API changes.
+
+Back-end part is primarily
+an internal feature, detailed implementation notes will remain in this RFC
+and in the API documentation.
+
+On front-end:
+- For developers, a documentation on how to build a custom Files UI plugin is probably desirable.
+- For site admins, we need to document the `FILES_UI_ENTRYPOINT` config option.
+
+## Drawbacks
+
+- A slightly more complicated code.
+- Some files UI components and Redux store actions that were private now need to be exported to allow reuse.
+- Reliance on the global state (`window.invenio.files`) for registering the Files UI, along with an additional Webpack entry point that must be included in a specific order.
+- Need to create a Webpack entrypoint in site's webpack to be able to use any custom Files UI implementations, that are just NPM packages (like the Uppy.io integration above).
+
+## Alternatives
+
+Having separate API endpoints for handling the use cases above. This RFC provides a cleaner solution.
+
+For the global `window.invenio drawback`, a prop-based approach could potentially be used to pass the correct services to the form app. The `FilesUploader` field could be provided
+via `react-overridable` by mapping the desired component in `mapping.js`. That would avoid the need for an extra webpack entrypoint and using the global state altogether. However,
+it remains unclear how site administrators could easily configure the correct services server-side (e.g., DepositFilesService, DepositFilesAPIClient), as they are purely JavaScript objects.
```

Comment:
Agree, often when we want to translate Flask/Python config to more dynamic JS functionality, we lose a lot of flexibility, and usually have to start double-keeping config-key-to-JS-object "registries".